src/main/java/softdesign/map/CartesianGridMap.java

```diff
@@ -1,50 +1,85 @@
 package main.java.softdesign.map;
 
 import javax.vecmath.Point3d;
+import main.java.softdesign.Environment;
 
 public class CartesianGridMap extends AbstractMap {
 
-	private final int cartesianOffset;
-
 	private Tile[][] grid;
+	private int coveredCount = 0;
 
-	public CartesianGridMap(int size) {
-		super(size);
-
-		this.cartesianOffset = -(this.size / 2);
+	public CartesianGridMap() {
+		super();
 
-		int axisPoints = size + 1;
+		int axisPoints = Environment.SIZE;
```

Comment:
This introduces tight coupling. The previous iteration is better. We should not be using the constant like this.